web/src/main/java/se/inera/intyg/webcert/web/csintegration/integration/ListQuestionConverter.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2024 Inera AB (http://www.inera.se)
+ *
+ * This file is part of sklintyg (https://github.com/sklintyg).
+ *
+ * sklintyg is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * sklintyg is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package se.inera.intyg.webcert.web.csintegration.integration;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+import se.inera.intyg.common.support.facade.model.question.Question;
+import se.inera.intyg.common.support.facade.model.question.QuestionType;
+import se.inera.intyg.common.support.facade.model.question.Reminder;
+import se.inera.intyg.webcert.persistence.arende.model.ArendeAmne;
+import se.inera.intyg.webcert.web.web.controller.api.dto.ArendeListItem;
+import se.inera.intyg.webcert.web.web.controller.facade.dto.CertificateDTO;
+
+@Component
+public class ListQuestionConverter {
+
+    public List<ArendeListItem> convert(Optional<CertificateDTO> certificate, Question question) {
+        if (certificate.isEmpty()) {
+            throw new IllegalStateException("Certificate for question was not available");
+        }
+        final var convertedQuestion = convertArendeListItem(certificate.get(), question);
+        final var convertedReminders = Arrays.stream(question.getReminders())
+            .map(reminder -> convertArendeListItem(certificate.get(), reminder))
+            .collect(Collectors.toList());
+
+        convertedReminders.add(convertedQuestion);
+
+        return convertedReminders;
+    }
+
+    private ArendeListItem convertArendeListItem(CertificateDTO certificate, String subject, String author, LocalDateTime sent,
```

Comment:
Don't we need to add status, signDate, forwarded and resourcelinks also to the ArendeListItem?

Reply:
Which value do you mean for signDate?

Reply:
Sorry, ignore signDate. I think I mixed up the conversion logic in CertificateListItemConverterImpl. 😅